### PR TITLE
Improved Excel button

### LINKED ISSUE #96: Allow for excel / csv export of listings
A lot of users want to download the data shown in the app. Wondering if there are any objections to adding a csv or excel button to all the `DT` tables?

### PR DESCRIPTION
**Pull request by aclark02-arcus**
Closes #96. This PR is currently an initial placeholder that generates the most basic of excel buttons using the "Buttons" extension for `{DT}`.

Some possible enhancements include:
- [x] styling and/or repositioning the button
    - Moved button to right-hand side
- [x] making the button disappear or disabled when no data exists in the table
    - Button only shows up on common forms & study forms, plus the queries table
- [x] Add excel button option to golem config
- [ ] styling the output in excel to maintain **bold** (new / updated) records
    - Seems tricky ([Stack Overflow](https://stackoverflow.com/questions/49069476/keep-formatting-when-exporting-table-with-dt-datatables-buttons-extension)) so I'll leave this out for now.
- [x] Improving the the specificity of the filename (Right now, all tables output a generic name).
    - See https://github.com/openpharma/clinsight/pull/122#issuecomment-2455167310

### CODE REVIEW

**Comment by LDSamson** [comment]
It would be nice if we could reposition it is not the primary visual that people should look at when being presented by the table. Users will look for it if needed. Should we also disable it by default and only enable in the tables in which it is needed? Or add a toggle in the app's  settings `enable download excel buttons` or something similar?

**Comment by aclark02-arcus** [comment]
View of filename's when downloading listings from different forms is below. Naming convention is as follows:

    {"clinsight", [form], [patient], [timestamp]}

Notice that if "show all patients" is selected, the filename will reflect this.

image

**Comment by aclark02-arcus** [comment]
Moved button to right-hand side so that it's not the first thing you see:

image

**Comment by aclark02-arcus** [comment]
@jthompson-arcus, got the coveted ✅

**Review by jthompson-arcus — COMMENTED** [review]
Few comments below.

File: R/mod_queries.R
[the remark this answers is not in the document]
Comment (aclark02-arcus) [reply R/mod_queries.R:1]: eh, `selected_query` is just a log file of sorts... didn't seem like something someone would want to download.

image

File: inst/golem-config.yml
[the remark this answers is not in the document]
Comment (aclark02-arcus) [reply inst/golem-config.yml:1]: reverted

File: R/mod_queries.R
[the remark this answers is not in the document]
Comment (aclark02-arcus) [reply R/mod_queries.R:1]: after chatting with you off-line. Let's just leave it off for now, and we can easily add later if we need it.

**Pull request merged**